server/sandbox_run_linux.go

```diff
@@ -698,43 +699,84 @@ func (s *Server) runPodSandbox(ctx context.Context, req *types.RunPodSandboxRequ
 		}
 	}
 
-	overhead := sbox.Config().GetLinux().GetOverhead()
+	overhead := sbuilder.Config().GetLinux().GetOverhead()
 	overheadJSON, err := json.Marshal(overhead)
 	if err != nil {
 		return nil, err
 	}
 	g.AddAnnotation(annotations.PodLinuxOverhead, string(overheadJSON))
 
-	resources := sbox.Config().GetLinux().GetResources()
+	resources := sbuilder.Config().GetLinux().GetResources()
 	resourcesJSON, err := json.Marshal(resources)
 	if err != nil {
 		return nil, err
 	}
 	g.AddAnnotation(annotations.PodLinuxResources, string(resourcesJSON))
 
-	sb, err := libsandbox.New(sbox.ID(), namespace, sbox.Name(), kubeName, logDir, labels, kubeAnnotations, processLabel, mountLabel, metadata, shmPath, cgroupParent, privileged, runtimeHandler, sbox.ResolvPath(), hostname, portMappings, hostNetwork, created, usernsMode, overhead, resources)
-	if err != nil {
-		return nil, err
+	seccompRef := types.SecurityProfile_Unconfined.String()
+	if !privileged {
+		_, ref, err := s.config.Seccomp().Setup(
+			ctx,
+			s.config.SystemContext,
+			nil,
+			"",
+			"",
+			nil,
+			nil,
+			g,
+			securityContext.Seccomp,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("setup seccomp: %w", err)
+		}
+		seccompRef = ref
 	}
 
-	sb.SetDNSConfig(sbox.Config().DnsConfig)
+	hostnamePath := podContainer.RunDir + "/hostname"
+
+	sbuilder.SetCriSandbox(sandboxID, created, labels, kubeAnnotations, metadata)
+	sbuilder.SetNamespace(namespace)
+	sbuilder.SetName(sandboxName)
+	sbuilder.SetKubeName(kubeName)
+	sbuilder.SetLogDir(logDir)
+	sbuilder.SetContainers(memorystore.New[*oci.Container]())
+	sbuilder.SetProcessLabel(processLabel)
+	sbuilder.SetMountLabel(mountLabel)
+	sbuilder.SetShmPath(shmPath)
+	sbuilder.SetCgroupParent(cgroupParent)
+	sbuilder.SetPrivileged(privileged)
+	sbuilder.SetRuntimeHandler(runtimeHandler)
+	sbuilder.SetResolvPath(sbuilder.ResolvPath())
+	sbuilder.SetHostname(hostname)
+	sbuilder.SetPortMappings(portMappings)
+	sbuilder.SetHostNetwork(hostNetwork)
+	sbuilder.SetUsernsMode(usernsMode)
+	sbuilder.SetPodLinuxOverhead(overhead)
+	sbuilder.SetPodLinuxResources(resources)
+	sbuilder.SetDNSConfig(sbuilder.Config().DnsConfig)
+	sbuilder.SetHostnamePath(hostnamePath)
+	sbuilder.SetNamespaceOptions(securityContext.NamespaceOptions)
+	sbuilder.SetSeccompProfilePath(seccompRef)
+	sbuilder.SetCreatedAt(created)
```

Comment:
I think these steps should happen incrementally throughout, though we can leave them like this for now if it's helpful to make sure we're calling everything (before we have validation on the sbuilder object before it spits out a sandbox)